### PR TITLE
named export in system-task, using require

### PR DESCRIPTION
**Pull request by aminya**
- using normal `require` instead of dynamic `require`. 
`require.resolve` is slower and is not a first-class function and should be avoided in non-remote situations.
https://www.bennadel.com/blog/3243-using-require-resolve-to-calculate-module-relative-file-paths-in-node-js.htm
> Clearly, the require.resolve() method is doing far more than just generating file paths - it's actually interacting with the file system. As such, it has greater processing overhead and some limitations in the way it can be used.
- This removes the errors that occur when the package is used with a compiler.

### CODE REVIEW

File: lib/system-task.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const OP_DATA = 0b001;
 const OP_STAT = 0b010;
 const OP_REAL = 0b100;
 
-module.exports.op = {
+const op = {
```
Comment (Alhadis) [inline lib/system-task.js:17]: Why was this changed?

File: lib/system.js [hunk 2]
```diff
@@ -1,9 +1,7 @@
 "use strict";
 
 const {Task}     = require("atom");
-const taskPath   = require.resolve("./system-task.js");
-const {op}       = require(taskPath);
-
+const {op, taskPath} = require("./system-task")
```
Comment (Alhadis) [inline lib/system.js:4]: Please use semicolons.

File: lib/system-task.js [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 const fs = require("fs");
 
-module.exports = function(jobs){
+module.exports.taskPath = function taskPath(jobs){
```
Comment (Alhadis) [inline lib/system-task.js:6]: Exporting a function is a requirement of [Atom's `Task` API](https://flight-manual.atom.io/api/v1.46.0/Task/)...

Comment (aminya) [reply lib/system-task.js:6]: I don't think this is a requirement. `Atom's` task API is very different anyways. We don't need to follow it.

File: lib/system-task.js (hunk 1 above)
Comment (aminya) [reply lib/system-task.js]: Fixed in https://github.com/file-icons/fs/pull/8/commits/7b6329b9ffeee2702883b9bee92322ad101a676d

**Comment by aminya** [comment]
Using a technique from [here](https://www.bignerdranch.com/blog/default-exports-or-named-exports-why-not-both/), I kept the backward compatible default export but added a named export for its benefits.

File: lib/system-task.js (hunk 3 above)
Comment (aminya) [reply lib/system-task.js]: Fixed in https://github.com/file-icons/fs/pull/8/commits/5c1f55f858343866a123c1e5a211486d050f2522

File: lib/system.js (hunk 2 above)
Comment (aminya) [reply lib/system.js]: Fixed in https://github.com/file-icons/fs/pull/8/commits/5c1f55f858343866a123c1e5a211486d050f2522

File: lib/system-task.js (hunk 3 above)
Comment (Alhadis) [reply lib/system-task.js]: What are you talking about? This is the file that's called by Atom's `Task` API, which [as you can see](https://github.com/atom/atom/blob/166e7d23867760b9881004143df57232c14bdf6b/src/task.coffee), takes a path to a file that's expected to export a function as its default export. Which is a seriously stupid design design, but hey, Atom are undisputed champions of bad APIs.

Comment (aminya) [reply lib/system-task.js]: I didn't see that one. I tested the code with named export and works without any issues. That is not a requirement.

**Comment by Alhadis** [comment]
Have you actually *tested* your changes? Because I really think you should:

Figure 1

The "export `default`" trick won't work here because Atom is using the older CommonJS module system.

I still fail to see why these changes are necessary.

**Comment by aminya** [comment]
Try again. I moved the code to the end of the file.

File: lib/system-task.js (hunk 3 above)
Comment (Alhadis) [reply lib/system-task.js]: I don't know how you're testing this, but you should have seen clear breakage at startup:

Figure 2

Again, could you **please** explain why you think these changes are necessary?

Comment (aminya) [reply lib/system-task.js]: This is strange. It works on my Atom. The main thing of this PR was the removal of `require.resolve`. I can bring back the default export if does not work for you.

Comment (Alhadis) [reply lib/system-task.js]: I repeat, **what are you trying to achieve with this pull-request**? Do you even know the difference between `require.resolve` and `require`? Or even what `require.resolve` does?

I'm so confused as to what your goal is.

Comment (aminya) [reply lib/system-task.js]: - `require.resolve` is slower.
- It doesn't work with compilers.

Comment (Alhadis) [reply lib/system-task.js]: > `require.resolve` is slower

\*sigh\* What the author of that [blog post](https://www.bennadel.com/blog/3243-using-require-resolve-to-calculate-module-relative-file-paths-in-node-js.htm) meant was that resolving a path *statically* is faster than resolving a path based on module resolution. Which is a given. But `require.resolve` is no slower than `require` because [it's used internally by the latter](https://nodejs.org/api/modules.html#modules_require_resolve_request_options):

> Use the internal `require()` machinery to look up the location of a module, but rather than loading the module, just return the resolved filename.

As for compilers, that's an entirely different issue, and more of a problem with whatever bundler you're using than it is with codebase structure. You haven't explained anything about what you're trying to bundle this for, so I'm at a loss as to how to help here.

Comment (aminya) [reply lib/system-task.js]: Anyways, I don't think we need to use `require.resolve` here when we clearly know what we want to `require`. Let's remove that overhead.

Comment (Alhadis) [reply lib/system-task.js]: Have you been paying attention to anything I've said? I already explained that Atom's `Task` API (a *class*) takes a *filesystem path to a module*, and **IT** does the `require` work. Internally. Inside Atom's core. Places I have no control or power over.

Understand now?

Comment (aminya) [reply lib/system-task.js]: What do you mean? It works now. What is the problem? I don't understand why you insist on using something with more overhead when the new suggestion works perfectly fine.

Comment (Alhadis) [reply lib/system-task.js]: > You just need to path a string to `Task.once`

Holy fucking shit, what do you think `require.resolve` returns?

Look. I've wasted an exorbitant amount of time trying to explain this to you. Either you don't understand, and/or you're not listening to a thing I'm saying.

Comment (aminya) [reply lib/system-task.js]: First of all, mind your tone.
Second, I am the one who is wasting his time fixing your work.

Comment (Alhadis) [reply lib/system-task.js]: Right, I've hit breaking point. Consider yourself banned.

Fuck off already.
